Automattic/wp-calypso
# Editor/CPT: Refactor status loading label for non-hardcoded types

This pull request seeks to refactor the `<EditorStatusLabel />` component to accommodate custom post types. Additionally, it moves the current component from the root `client/post-editor` directory such that it's treated as an independent component much the same as other editor components.

__Testing instructions:__

Verify that no regressions exist with the status label displaying the current post status or with the status label displaying "Loading Post…" or "Loading Page…" for posts and pages respectively. Because custom post types are not currently enabled for editing, it's a bit trickier to test the label loading behavior. To do so, apply the following changes to your local environment and restart the current process with `DISABLE_FEATURES=persist-redux make run`:

```diff
diff --git a/client/post-editor/editor-status-label/placeholder.jsx b/client/post-editor/editor-status-label/placeholder.jsx
index 6892892..9461a3b 100644
--- a/client/post-editor/editor-status-label/placeholder.jsx
+++ b/client/post-editor/editor-status-label/placeholder.jsx
@@ -21,16 +21,12 @@ function EditorStatusLabelPlaceholder( { siteId, typeSlug, type, className } ) {
 
        let postType;
        switch ( typeSlug ) {
-               case 'post': postType = translate( 'Post', { context: 'noun' } ); break;
-               case 'page': postType = translate( 'Page' ); break;
                default: postType = get( type, 'labels.singular_name', '' );
        }
 
        return (
                <button className={ classes }>
-                       { 'post' !== typeSlug && 'page' !== typeSlug && (
-                               <QueryPostTypes siteId={ siteId } />
-                       ) }
+                       <QueryPostTypes siteId={ siteId } />
                        <strong>
                                { translate( 'Loading %(postType)s…', { args: { postType } } ) }
                        </strong>
``` 

1. Navigate to the [Calypso post editor](http://calypso.localhost:3000/post)
2. Select a site, if prompted
3. Click the Drafts button in the editor sidebar, or create and save a new draft if no drafts exist
4. Click a draft to load it
5. Note that a "Loading Post…" or "Loading Page…" text is shown as the status label until the post or page loads
